avro-codegen/src/main/java/com/linkedin/avroutil1/codegen/SpecificRecordGeneratorUtil.java

```diff
@@ -249,6 +249,54 @@ public static boolean isMapTransformerApplicable(AvroSchema schema) {
     return isNullUnionOf(AvroType.MAP, schema);
   }
 
+  /**
+   * If a schema is primitive type or a null union of primitive type, it can be handled as a primitive type
+   * @param schema schema to be validated
+   * @return true if schema can be handled as primitive, else false
+   */
+  private static boolean canBeHandledAsPrimitiveType(AvroSchema schema) {
+    if (schema instanceof AvroUnionSchema) {
+      if (((AvroUnionSchema) schema).getTypes().size() != 2) {
```

Comment:
if `getTypes().size() == 1`, couldn't it still be handled as a primitive?

Reply:
unions of size 1 are not nullable. For example, a union of array is not a nullable array. So we couldn't handle that case